### PR TITLE
[posix] add UART flush and clean up

### PR DESCRIPTION
**Pull request by bukepo**
This commit mainly implements the `otPlatUartFlush()`.

* fix code because MSG_NOSIGNAL is not a macro.
* set SO_NOSIGPIPE once.

### CODE REVIEW

File: src/posix/platform/uart.cpp [hunk 1]
```diff
@@ -328,55 +377,64 @@ void platformUartProcess(const fd_set *aReadFdSet, const fd_set *aWriteFdSet, co
...
+
+exit:
+    return;
+}
+
+otError otPlatUartFlush(void)
+{
+    otError error = OT_ERROR_NONE;
+
+    while (sWriteLength > 0)
```
Comment (gjc13) [inline src/posix/platform/uart.cpp:400]: Should `sWriteLength` be updated after flush?

Comment (bukepo) [reply src/posix/platform/uart.cpp:400]: it's updated in function platformUartWrite().

**Comment by bukepo** [comment]
@d1ee could you try whether this PR fixes the issue in #5310?

**Comment by d1ee** [comment]
@bukepo Sure, I'll test it and let you know.

**Comment by d1ee** [comment]
@bukepo I tested it and it works. Thanks!

**Pull request merged**